python-package/lightgbm/basic.py

```diff
@@ -786,7 +786,7 @@ def _data_from_pandas(
     feature_name: _LGBM_FeatureNameConfiguration,
     categorical_feature: _LGBM_CategoricalFeatureConfiguration,
     pandas_categorical: Optional[List[List]]
-) -> Tuple[np.ndarray, List[str], List[str], List[List]]:
+) -> Tuple[np.ndarray, List[str], Union[List[str], List[int]], List[List]]:
```

Comment:
This hint was wrong.

`categorical_feature` can be a list of integers (indicating 0-based column positions).

https://github.com/microsoft/LightGBM/blob/b74528f3f7b5b2731920afb09ebd41e8f52fc976/python-package/lightgbm/basic.py#L1724-L1726

And it's passed back out of this function unmodified if it's provided like that.

https://github.com/microsoft/LightGBM/blob/b74528f3f7b5b2731920afb09ebd41e8f52fc976/python-package/lightgbm/basic.py#L814-L817